include/drivers/i2c.h

```diff
@@ -121,6 +133,61 @@ struct i2c_msg {
 	u8_t		flags;
 };
 
+/** @brief Macro for initializing set of messages to perform TX followed by RX
+ *	   transfers.
+ * @param txbuf TX buffer.
+ * @param txlen TX data length.
+ * @param rxbuf RX buffer.
+ * @param rxlen RX data length.
+ * @param delay Delay (in milliseconds) before starting given TX if executed
+ *		in sequence.
+ */
+#define I2C_MSG_TXRX(txbuf, txlen, rxbuf, rxlen, delay) \
+	{ \
+		.buf = (u8_t *)txbuf, \
+		.len = txlen, \
+		.flags = I2C_MSG_WRITE | I2C_MSG_DELAY(delay) \
+	}, \
+	{ \
+		.buf = (u8_t *)rxbuf, \
+		.len = rxlen, \
+		.flags = I2C_MSG_RESTART | I2C_MSG_READ | I2C_MSG_STOP \
+	}
+
+/** @brief Macro for initializing set of messages to perform TX followed by TX
+ *	   transfers.
+ * @param txbuf0 First TX buffer.
+ * @param txlen0 First TX data length.
+ * @param txbuf1 Second TX buffer.
+ * @param txlen1 Second TX data length.
+ * @param delay Delay (in milliseconds) before starting TX if executed in
+ *		sequence.
+ */
+#define I2C_MSG_TXTX(txbuf0, txlen0, txbuf1, txlen1, delay) \
+	{ \
+		.buf = (u8_t *)txbuf0, \
+		.len = txlen0, \
+		.flags = I2C_MSG_WRITE | I2C_MSG_DELAY(delay) \
+	}, \
+	{ \
+		.buf = (u8_t *)txbuf1, \
+		.len = txlen1, \
+		.flags = I2C_MSG_WRITE | I2C_MSG_STOP \
+	}
+
+/** @brief Macro for initializing TX message
+ * @param txbuf TX buffer.
+ * @param txlen TX data length.
+ * @param delay Delay (in milliseconds) before starting TX if executed in
+ *		sequence.
+ */
+#define I2C_MSG_TX(txbuf, txlen, delay) \
```

Comment:
Could you have a separate patch for all the above? (Am I wrong if I say it could be used on i2c_transfer()? thus why I think it deserves its own patch.)

Reply:
yes, except for delay parameter which was added to the flags. but you are right, i will extract it to separate patch.